src/tactic/lint.lean

```diff
@@ -524,6 +524,28 @@ some <$> do
   errors_found := "LEFT-HAND SIDE NOT IN SIMP-NF.\n" ++
     "Some simp lemmas have a left-hand side that is not in simp-normal form." }
 
+private meta def simp_var_head (d : declaration) : tactic (option string) := do
+tt ← is_simp_lemma d.to_name | pure none,
+-- Sometimes, a definition is tagged @[simp] to add the equational lemmas to the simp set.
+-- In this case, ignore the declaration if it is not a valid simp lemma by itself.
+tt ← is_valid_simp_lemma_cnst d.to_name | pure none,
+lhs ← simp_lhs d.type,
+head_sym@(expr.local_const _ _ _ _) ← pure lhs.get_app_fn | pure none,
+pure <$> do
```

Comment:
```suggestion
```
Am I misreading, or is this unnecessary?

Reply:
```lean
lemma pure_some {α} (a : α) : pure a = some a := rfl
```
:smile:

Reply:
But with the coercion `T -> option T`, this works (slightly differently, but equally well) without the `pure <$> do` line :)

Reply:
Oh, you're right.